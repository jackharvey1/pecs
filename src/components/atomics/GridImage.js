import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ResponsiveImage = styled.img`
    --width: calc(100vw/3);

    @media (min-width: 768px) {
        --width: calc(100vw/4);
    }

    @media (min-width: 1024px) {
        --width: calc(100vw/6);
    }

    @media (min-width: 1280px) {
        --width: calc(100vw/8);
    }

    height: var(--width);
    width: var(--width);
`;

const GridImage = ({ src, alt, addToStoryboard }) => (
    <ResponsiveImage
        src={src}
        alt={alt}
        onClick={() => addToStoryboard(src, alt)}
    />
);

GridImage.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
    addToStoryboard: PropTypes.func.isRequired,
};

export default GridImage;
