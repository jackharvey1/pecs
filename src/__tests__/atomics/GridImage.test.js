import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GridImage from '../../components/atomics/GridImage';

describe('GridImage component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(
            <GridImage
                src="http://image.com/image.png"
                alt="image"
                addToStoryboard={() => null}
            />
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('calls addToStoryboard with src and alt on click', () => {
        const mockAddToStoryboard = jest.fn();

        const wrapper = shallow(
            <GridImage
                src="http://image.com/image.png"
                alt="image"
                addToStoryboard={mockAddToStoryboard}
            />
        );

        wrapper.simulate('click');

        expect(mockAddToStoryboard).toHaveBeenCalledWith('http://image.com/image.png', 'image');
    });
});
