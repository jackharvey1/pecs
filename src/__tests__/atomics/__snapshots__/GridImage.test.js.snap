// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridImage component renders as intended 1`] = `
.c0 {
  --width: calc(100vw/3);
  height: var(--width);
  width: var(--width);
}

@media (min-width:768px) {
  .c0 {
    --width: calc(100vw/4);
  }
}

@media (min-width:1024px) {
  .c0 {
    --width: calc(100vw/6);
  }
}

@media (min-width:1280px) {
  .c0 {
    --width: calc(100vw/8);
  }
}

<img
  alt="image"
  className="c0"
  onClick={[Function]}
  src="http://image.com/image.png"
/>
`;
